### ISSUE TITLE
Robusting the demos

### ISSUE THREAD

**Issue by sruloart** [HaxeFlixel/flixel-demos#88]
I think that a lot of HaxeFlixel's features are basically hidden from sight, since both the Handbook and the API are not structured like answers to common questions, i.e. if I ask: "How can I do this?" the answer for now is: "This class can do this and that". Well, thank you computer, but, let's say I don't know what class should I use, and how can I implement it (and even if I should). Now what?

 The answer should be: "Here, look, there are multiple ways you can achieve this, these are the pros and cons of each way". The forum, with all of its problem-solving might, usually just answers user-specific questions, and it's also not the proper way you should teach someone anything, IMHO. So, along come the demos. 

I suggest (to myself and others) making a lot more of the feature-based demos, introducing only one common and specific feature and its variations, bareboned, with a tutorial-like comments inside of each such project. Remember, It's not about describing a class, it's about describing possible ways to accomplish a common game-related task. 

tl;dr: I think we should focus not on what the framework can do, but on the common things people want to do using it...
